src/IdealizedMesoscaleExperiments/eddying_channel.jl

```diff
@@ -0,0 +1,301 @@
+        buoyancy = BuoyancyTracer(),
+        coriolis = coriolis,
+        closure = (horizontal_diffusive_closure, vertical_diffusive_closure, convective_adjustment),
+        tracers = (:b),
+        boundary_conditions = (b = b_bcs, u = u_bcs, v = v_bcs),
+        forcing = (; b = Fb))
+
+    @info "Built $model."
+
+    #####
+    ##### Initial conditions
+    #####
+
+    # resting initial condition
+    ε(σ) = σ * randn()
+    bᵢ(x, y, z) = parameters.ΔB * (exp(z / parameters.h) - exp(-Lz / parameters.h)) / (1 - exp(-Lz / parameters.h)) + ε(1e-8)
+    uᵢ(x, y, z) = ε(1e-8)
+    vᵢ(x, y, z) = ε(1e-8)
+    wᵢ(x, y, z) = ε(1e-8)
+
+    Δy = 100kilometers
+    Δz = 100
+
+    set!(model, b = bᵢ, u = uᵢ, v = vᵢ, w = wᵢ)
+
+    #####
+    ##### Simulation building
+    #####
+
+    simulation = Simulation(model, Δt = Δt₀, stop_time = stop_time)
+
+    # add timestep wizard callback
+    wizard = TimeStepWizard(cfl = 0.1, max_change = 1.1, max_Δt = 20minutes)
+    simulation.callbacks[:wizard] = Callback(wizard, IterationInterval(20))
+
+    # add progress callback
+    wall_clock = [time_ns()]
+
+    function print_progress(sim)
+        @printf("[%05.2f%%] i: %d, t: %s, wall time: %s, max(u): (%6.3e, %6.3e, %6.3e) m/s, next Δt: %s\n",
+            100 * (sim.model.clock.time / sim.stop_time),
+            sim.model.clock.iteration,
+            prettytime(sim.model.clock.time),
+            prettytime(1e-9 * (time_ns() - wall_clock[1])),
+            maximum(abs, sim.model.velocities.u),
+            maximum(abs, sim.model.velocities.v),
+            maximum(abs, sim.model.velocities.w),
+            prettytime(sim.Δt))
+
+        wall_clock[1] = time_ns()
+
+        return nothing
+    end
+
+    simulation.callbacks[:print_progress] = Callback(print_progress, IterationInterval(20))
+
+
+    #####
+    ##### Diagnostics
+    #####
+
+    u, v, w = model.velocities
+    b = model.tracers.b
+    η = model.free_surface.η
+
+    ζ = Field(∂x(v) - ∂y(u))
+
+    B = Field(Average(b, dims = 1))
+    U = Field(Average(u, dims = 1))
+    η̄ = Field(Average(η, dims = 1))
+    V = Field(Average(v, dims = 1))
+    W = Field(Average(w, dims = 1))
+
+    b′ = b - B
+    u′ = u - U
+    v′ = v - V
+    w′ = w - W
+
+    tke_op = @at (Center, Center, Center) (u′ * u′ + v′ * v′ + w′ * w′) / 2
+    tke = Field(Average(tke_op, dims = 1))
+
+    uv_op = @at (Center, Center, Center) u′ * v′
+    vw_op = @at (Center, Center, Center) v′ * w′
+    uw_op = @at (Center, Center, Center) u′ * w′
+
+    u′v′ = Field(Average(uv_op, dims = 1))
+    v′w′ = Field(Average(vw_op, dims = 1))
+    u′w′ = Field(Average(uw_op, dims = 1))
+
+    b′b′ = Field(Average(b′ * b′, dims = 1))
+    v′b′ = Field(Average(b′ * v′, dims = 1))
+    w′b′ = Field(Average(b′ * w′, dims = 1))
+
+    outputs = (; b, ζ, u, v, w)
+
+    zonally_averaged_outputs = (b = B, u = U, v = V, w = W, η = η̄,
+        vb = v′b′, wb = w′b′, bb = b′b′,
+        tke = tke, uv = u′v′, vw = v′w′, uw = u′w′)
```

Comment:
It doesn't look like there's any output writer for these quantities. We need the option for both these quantities and time-averages of these quantities.